README.md

```diff
@@ -487,6 +487,48 @@ static void BM_ManualTiming(benchmark::State& state) {
 BENCHMARK(BM_ManualTiming)->Range(1, 1<<17)->UseManualTime();
 ```
 
+There is also a second overload of `UseManualTime`, which allows the user to pass an arbitrary unit name, an optional scaling multiplier for those units, and a way to convert those units to a cost that will be used by Benchmark internally to determine maximum iterations/initial iteration cost and similar:
+
+```c++
+#include <cstddef>
+
+#if _WIN32
+#include <intrin.h>
+#else
+#include <x86intrin.h>
+#endif
+
+int64_t rdtsc() {
+  return static_cast<int64_t>(__rdtsc());
+}
+
+static double compute_cost(double measurement) {
+  // convert to some representative cost
+  return measurement;
+}
+
+static void BM_ManualTimingWithUnits(benchmark::State& state) {
+  int rotations = state.range(0);
+  std::unique_ptr<int32_t[]> data(new int32_t[rotations]]);
+
+  for (auto _ : state) {
```

Comment:
I think this example is an antipattern. It should use `KeepRunningBatch`, no?
Starting and stopping the timer every iteration certainly is. We don't recommend writing benchmarks like this, we certainly shouldn't provide them as examples.

The library goes out of it's way to give you a better way to do this.

Reply:
I'm a bit confused. The Readme specifically calls out using the `for` loop syntax as a better KeepRunning loop. Maybe I just lack understanding of how `KeepRunningBatch` is meant to be used to make this scenario better? `Batch` and `KeepRunningBatch` don't even exist on the readme... am I missing something? Is this a v2 feature?

We specify `UseManualTime` here; this means we have to call `SetIterationTime()` each iteration, that's why I'm using the manual timing here. I wish there was a way to batch-measure, but there is no `SetRunTime(total)` that divides by the # of iterations internally like it does for the non-manual measure.

Reply:
I can see why that documentation might be confusing.

The `KeepRunningBatch` documentation is found in the header, here: https://github.com/google/benchmark/blob/master/include/benchmark/benchmark.h#L481

The `for (auto)` formulation is preferred as default. It ekes out every bit of optimization possible. When you need to do more complicated things, like batching, you need to use other interfaces.

Everything around the main benchmark loop should be as optimized as possible. Otherwise it introduces noise into the benchmark results.

Reply:
I changed the benchmark to be what I think is more appropriate for a rotl measurement, using `KeepRunningBatch`. This serves to benchmark both `KeepRunningBatch` and the advanced `UseManualTime`, but it probably doesn't address your other concerns.